src/table/TablePaginationActions.jsx

```diff
@@ -0,0 +1,90 @@
+
+import IconButton from '@material-ui/core/IconButton';
+import FirstPageIcon from '@material-ui/icons/FirstPage';
+import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
+import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
+import LastPageIcon from '@material-ui/icons/LastPage';
+import { makeStyles } from '@material-ui/core/styles';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const useStyles = makeStyles({
+  root: {
+    flexShrink: 0,
+  },
+  disabled: {
+    color: 'rgba(0, 0, 0, 0.3)'
+  }
+});
+
+export default function TablePaginationActions(props) {
+  const classes = useStyles();
+  const { count, page, rowsPerPage, onChangePage } = props;
+
+  const handleFirstPageButtonClick = (event) => {
+    onChangePage(event, 0);
+  };
+
+  const handleBackButtonClick = (event) => {
+    onChangePage(event, page - 1);
+  };
+
+  const handleNextButtonClick = (event) => {
+    onChangePage(event, page + 1);
+  };
+
+  const handleLastPageButtonClick = (event) => {
+    onChangePage(event, Math.ceil(count / rowsPerPage) - 1);
+  };
+
+  const onFirstPage = page === 0;
+  const onLastPage = page >= Math.ceil(count / rowsPerPage) - 1;
+
+  return (
+    <div className={classes.root}>
+      <IconButton 
+        onClick={!onFirstPage && handleFirstPageButtonClick}
+        aria-disabled={onFirstPage}
+        aria-label="first page"
+        title="First page"
+        className={onFirstPage && classes.disabled}
+      >
+        <FirstPageIcon />
+      </IconButton>
+      <IconButton 
+        onClick={!onFirstPage && handleBackButtonClick}
+        aria-disabled={onFirstPage}
+        aria-label="previous page"
+        title="Previous page"
+        className={onFirstPage && classes.disabled}
+      >
+        <KeyboardArrowLeft />
+      </IconButton>
+      <IconButton 
+        onClick={!onLastPage && handleNextButtonClick}
+        aria-disabled={onLastPage}
+        aria-label="next page"
+        title="Next page"
+        className={onLastPage && classes.disabled}
+      >
+        <KeyboardArrowRight />
+      </IconButton>
+      <IconButton 
+        onClick={!onLastPage && handleLastPageButtonClick}
+        aria-disabled={onLastPage}
+        aria-label="last page"
+        title="Last page"
+        className={onLastPage && classes.disabled}
+      >
+        <LastPageIcon />
+      </IconButton>
+    </div>
```

Comment:
one thing that always comes handy in this kind of situations is using an array of (configuration) objects instead of putting multiple `<IconButton />` element in our jsx files. the basic idea is about preventing large module files. 

if we could have an aray of configs for each button like below: 
```
const tableBtnsArr = [
  {
    order: 0,
    IconProps: {
      onClick: !onLastPage && handleLastPageButtonClick,
      'aria-disabled' : onLastPage,
      'aria-label': 'last page',
      title: 'Last Page',
      className: onLastPage && classes.disabled,
    },
    IconToRender: <LastPageIcon />
  }, 
  { ...another button here },
  { ...another button here },
].sort((a, b) => a.order - b.order)
```

and then in your return clause you can do:
```
tableBtnsArr.map(btn => (
    <IconButton 
        key={btn.order}
        {...btn.IconProps}
      >
         {btn.IconToRender}
      </IconButton>
))
```

this will prevent from large component files all the time and fix the problem for code climate as well.

Reply:
Every object in the array will be similar and you might get the code climate problem anyway. We shouldn't worry too much about the similarity issues, but when it is identical you should most likely refactor.

Reply:
yeah, I think leaving the jsx as it is in this case will be clearer, but I'll keep it in mind :)

Reply:
rather than that, it's looking fine to me 👌